src/libraries/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpRequestCallback.cs

```diff
@@ -20,6 +24,8 @@ internal static class WinHttpRequestCallback
         public static Interop.WinHttp.WINHTTP_STATUS_CALLBACK StaticCallbackDelegate =
             new Interop.WinHttp.WINHTTP_STATUS_CALLBACK(WinHttpCallback);
 
+        private static bool CertificateCachingAppContextSwitchEnabled = AppContext.TryGetSwitch("System.Net.Http.UseWinHttpCertificateCaching", out bool certificateCachingEnabled) && certificateCachingEnabled;
```

Comment:
```suggestion
        private static bool CertificateCachingAppContextSwitchEnabled { get; } = AppContext.TryGetSwitch("System.Net.Http.UseWinHttpCertificateCaching", out bool enabled) && enabled;
```
This makes it readonly and lets the JIT delete the unreachable code paths.